sources/ClangSharp.PInvokeGenerator/PInvokeGenerator.cs

```diff
@@ -1389,6 +1386,50 @@ private string GetTypeName(Cursor cursor, Cursor context, Type type, out string
             {
                 name = GetTypeNameForPointeeType(cursor, context, referenceType.PointeeType, out var nativePointeeTypeName);
             }
+            else if (type is TemplateSpecializationType templateSpecializationType)
+            {
+                var nameBuilder = new StringBuilder();
+
+                var templateTypeDecl = ((RecordType)templateSpecializationType.CanonicalType).Decl;
+                nameBuilder.Append(GetRemappedName(templateTypeDecl.Name, templateTypeDecl, tryRemapOperatorName: false));
+
+                nameBuilder.Append('<');
+
+                bool shouldWritePrecedingComma = false;
+
+                foreach (var arg in templateSpecializationType.Args)
+                {
+                    if (shouldWritePrecedingComma)
+                    {
+                        nameBuilder.Append(',');
+                        nameBuilder.Append(' ');
+
+                    }
+
+                    var typeName = GetRemappedTypeName(cursor, context: null, arg.AsType, out _);
+
+                    if (typeName == "bool")
+                    {
+                        // bool is not blittable, so we shouldn't use it for P/Invoke signatures
+                        typeName = "byte";
+                    }
+
+                    if (typeName.EndsWith("*"))
+                    {
+                        // Pointers are not yet supported as generic arguments; remap to UIntPtr
+                        typeName = "UIntPtr";
```

Comment:
Should this account for "latest" vs "compatible" codegen and use `nuint`?

Reply:
This `UIntPtr` is semantically equal to a `void*`, which doesn't provide any operations (like addition or subtraction) unless you cast to a different type of pointer. I don't think `nuint` would be an appropriate analog for that.

Reply:
In that case, most .NET handles are exposed as `IntPtr` and that is what most of the bindings in libraries like TerraFX or WinForms use.

Reply:
There also isn't a lot of difference, realistically speaking, between `UIntPtr` and `nuint`. Basically the only difference is that `nuint` supports the full set of operations and not just addition/subtraction with an `int`.

Reply:
I think `nuint` just makes less sense. It implies you are supposed to perform operations on the raw integer value (when you really shouldn't), and hides the idea the value is actually just a (type erased) reference to a memory location. If think this should be `IntPtr` instead, that is okay; however I figured that, since these are actual memory addresses, and not simply opaque handles, that the explicit unsigned makes more sense.

Reply:
Could you get an estimate of how much API surface would change in ClangSharp itself as well as something like TerraFX with the change to use `UIntPtr` as the default fallback type for pointers?

I'm namely concerned about this change: https://github.com/microsoft/ClangSharp/pull/192/files#diff-6a1f236ee26a1c65a174696f8f83527ae8fa16e6570aeedec19daec3ca6d70b4R1590 having widespread downstream impact and possibly usability concerns with .NET in general.

There is somewhat of a similar concern with representing `char` as `sbyte` by default considering that most of the UTF8 related APIs in .NET operate on `byte` instead, even though `sbyte` is more accurate to the MSVC/Clang defaults.

Reply:
Okay, I reverted the aforementioned change and this remaps to `IntPtr` now.